### PR TITLE
Add codes.merritt.Nyrna

### PR DESCRIPTION
**Pull request by Merrit**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/Merrit/nyrna
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: codes.merritt.Nyrna.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+app-id: codes.merritt.Nyrna
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
```
Comment (hfiguiere) [inline codes.merritt.Nyrna.yml:4]:
```suggestion
runtime-version: '22.08'
```

File: codes.merritt.Nyrna.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    tag: v2.9.1
+    # If flatpak-builder throws error `fatal: transport 'file' not allowed`
+    # during build resolve by running `git config --global --add protocol.file.allow always`
+    disable-shallow-clone: true
+  - type: file
+    path: codes.merritt.Nyrna.metainfo.xml
+- name: keybinder3
+  buildsystem: autotools
+  only-arches:
+  - x86_64
```
Comment (hfiguiere) [inline codes.merritt.Nyrna.yml:40]:
```suggestion
```

File: codes.merritt.Nyrna.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  post-install: install -Dm644 COPYING /app/share/licenses/keybinder-3/COPYING
```
Comment (hfiguiere) [inline codes.merritt.Nyrna.yml:41]:
```suggestion
  post-install: 
    - install -Dm644 COPYING /app/share/licenses/keybinder-3/COPYING
```

File: codes.merritt.Nyrna.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+- --share=ipc
+- --socket=fallback-x11
+- --socket=wayland
+- --device=dri
+- --socket=pulseaudio
+- --share=network
+- --talk-name=org.freedesktop.Flatpak # Access host processes
+modules:
+# Nyrna
+- name: nyrna
```
Comment (hfiguiere) [inline codes.merritt.Nyrna.yml:18]: since this is the main module it should be last.

File: codes.merritt.Nyrna.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    url: https://github.com/Merrit/nyrna/releases/download/v2.9.1/Nyrna-Linux-Portable.tar.gz
+    sha256: 896fea1ad94d19433d706e811e251292c1bea0c07153801152467cfe7500d715
+  - type: git
+    url: https://github.com/Merrit/nyrna.git
+    tag: v2.9.1
+    # If flatpak-builder throws error `fatal: transport 'file' not allowed`
+    # during build resolve by running `git config --global --add protocol.file.allow always`
+    disable-shallow-clone: true
+  - type: file
+    path: codes.merritt.Nyrna.metainfo.xml
```
Comment (hfiguiere) [inline codes.merritt.Nyrna.yml:36]: this file should be managed upstream for a future release.

**Comment by hfiguiere** [comment]
bot, build codes.merritt.Nyrna

File: codes.merritt.Nyrna.yml (hunk 5 above)
Comment (hfiguiere) [reply codes.merritt.Nyrna.yml:36]: actually it seems to be usptream already.

File: codes.merritt.Nyrna.yml (hunk 3 above)
Comment (hfiguiere) [reply codes.merritt.Nyrna.yml:41]:
```
(flatpak-builder:3): Json-WARNING **: 07:22:14.749: Failed to deserialize "post-install" property of type "GStrv" for an object of type "BuilderModule"
```

File: codes.merritt.Nyrna.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  buildsystem: autotools
+  only-arches:
+  - x86_64
+  post-install: install -Dm644 COPYING /app/share/licenses/keybinder-3/COPYING
+  sources:
+  - type: archive
+    url: https://github.com/kupferlauncher/keybinder/releases/download/keybinder-3.0-v0.3.2/keybinder-3.0-0.3.2.tar.gz
+    sha256: e6e3de4e1f3b201814a956ab8f16dfc8a262db1937ff1eee4d855365398c6020
+# libappindicator
+- shared-modules/libappindicator/libappindicator-gtk3-12.10.json
```
Comment (hfiguiere) [inline codes.merritt.Nyrna.yml:47]: `shared-modules` is missing.

**Comment by Merrit** [comment]
Made changes :+1:

**Comment by Merrit** [comment]
bot, build codes.merritt.Nyrna

**Comment by Merrit** [comment]
Looks like the build service had a network error?

**Comment by Merrit** [comment]
bot, build codes.merritt.Nyrna

**Comment by dginovker** [comment]
image

It'll be up soon

**Comment by barthalion** [comment]
bot, build codes.merritt.Nyrna

**Comment by barthalion** [comment]
You will need to submit your app to https://github.com/flathub/flatpak-builder-lint/blob/master/flatpak_builder_lint/staticfiles/exceptions.json. I assume you need access to flatpak-spawn to run SIGSTOP on arbitrary host processes?

**Comment by Merrit** [comment]
Exactly right, yes. I created a PR to add the exception, thank you for directing me to it! :)

**Comment by Merrit** [comment]
bot, build codes.merritt.Nyrna

**Comment by Merrit** [comment]
Seems to install and run as expected :+1:

**Comment by barthalion** [comment]
I get the "No windows that can be suspended" message on Fedora 37 Silverblue, despite having multiple windows open.

**Comment by Merrit** [comment]
Is that on Wayland? Only supports X11 at the moment unfortunately.

File: codes.merritt.Nyrna.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+app-id: codes.merritt.Nyrna
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: nyrna
+separate-locales: false
+finish-args:
+- --share=ipc
+- --socket=fallback-x11
+- --socket=wayland
```
Comment (hfiguiere) [inline codes.merritt.Nyrna.yml:11]: you just said it only support x11. so that's incorrect then.

Comment (Merrit) [reply codes.merritt.Nyrna.yml]: Thank you for pointing that out! <3

**Comment by Merrit** [comment]
bot, build codes.merritt.Nyrna

**Comment by Merrit** [comment]
Dependency check works inside Flatpak now. Also added a warning if a user tries to run it under Wayland, and changed the config to only specify X11 support. :+1:

**Pull request closed without merging**